scalalib/src/mill/javalib/android/AndroidSdkModule.scala

```diff
@@ -0,0 +1,175 @@
+   */
+  def ZipFileName: T[String] = T { "commandlinetools.zip" }
+
+  /**
+   * Path where the Android SDK will be installed.
+   *
+   * @return A `PathRef` representing the SDK installation directory.
+   */
+  def sdkDirectory: T[PathRef] = T { PathRef(millSourcePath / "android-sdk") }
+
+  /**
+   * Path to the Android SDK command-line tools directory.
+   *
+   * @return A `PathRef` representing the command-line tools directory.
+   */
+  def toolsDirectory: T[PathRef] = T { PathRef(sdkDirectory().path / ToolsDirName().toString) }
+
+  /**
+   * Path to the Android build tools based on the selected version.
+   *
+   * @return A `PathRef` representing the build tools directory.
+   *
+   * @see [[BuildToolsVersion]]
+   */
+  def buildToolsPath: T[PathRef] =
+    T { PathRef(sdkDirectory().path / "build-tools" / BuildToolsVersion().toString) }
+
+  /**
+   * Path to `android.jar`, required for compiling Android apps.
+   *
+   * @return A `PathRef` representing the path to `android.jar`.
+   *
+   * @see [[PlatformVersion]]
+   */
+  def androidJarPath: T[PathRef] =
+    T { PathRef(sdkDirectory().path / "platforms" / PlatformVersion().toString / "android.jar") }
+
+  /**
+   * Path to the D8 Dex compiler, used to convert Java bytecode to Dalvik bytecode.
+   *
+   * @return A `PathRef` representing the path to the D8 compiler.
+   *
+   * @see [[buildToolsPath]]
+   */
+  def d8Path: T[PathRef] = T { PathRef(buildToolsPath().path / "d8") }
+
+  /**
+   * Path to the Android Asset Packaging Tool (AAPT) for handling resources and packaging APKs.
+   *
+   * @return A `PathRef` representing the path to the AAPT tool.
+   *
+   * @see [[buildToolsPath]]
+   */
+  def aaptPath: T[PathRef] = T { PathRef(buildToolsPath().path / "aapt") }
+
+  /**
+   * Path to Zipalign, used to optimize APKs.
+   *
+   * @return A `PathRef` representing the path to the zipalign tool.
+   *
+   * @see [[buildToolsPath]]
+   */
+  def zipalignPath: T[PathRef] = T { PathRef(buildToolsPath().path / "zipalign") }
+
+  /**
+   * Path to the APK signer tool, used to sign APKs.
+   *
+   * @return A `PathRef` representing the path to the APK signer tool.
+   *
+   * @see [[buildToolsPath]]
+   */
+  def apksignerPath: T[PathRef] = T { PathRef(buildToolsPath().path / "apksigner") }
+
+  /**
+   * Installs the Android SDK by downloading the tools, extracting them,
+   * accepting licenses, and installing necessary components like platform and build tools.
+   *
+   * This method:
+   * - Downloads the SDK command-line tools from the specified URL.
+   *
+   * - Extracts the downloaded zip file into the specified SDK directory.
+   *
+   * - Accepts the SDK licenses required for use.
+   *
+   * - Installs essential components such as platform-tools, build-tools and platforms.
+   *
+   * @see [[SdkUrl]]
+   * @see [[toolsDirectory]]
+   * @see [[sdkDirectory]]
+   * @see [[BuildToolsVersion]]
+   * @see [[PlatformVersion]]
+   */
+  def installAndroidSdk: T[Unit] = T {
```

Comment:
This should return `T[PathRef]`. There should not be a `sdkDirectory`, `androidSdk` should download and unpack the SDK zip into its own `T.dest`

Reply:
done